plugins/security/nexus-ldap-plugin/src/main/java/org/sonatype/security/realms/ldap/internal/persist/orient/OrientDBLdapConfigurationSource.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.security.realms.ldap.internal.persist.orient;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.sonatype.nexus.component.services.id.EntityIdFactory;
+import org.sonatype.nexus.orient.DatabaseInstance;
+import org.sonatype.security.realms.ldap.internal.persist.LdapConfigurationSource;
+import org.sonatype.security.realms.ldap.internal.persist.entity.LdapConfiguration;
+import org.sonatype.sisu.goodies.lifecycle.LifecycleSupport;
+
+import com.google.common.collect.Lists;
+import com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx;
+import com.orientechnologies.orient.core.exception.OConcurrentModificationException;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+import com.orientechnologies.orient.core.storage.ORecordDuplicatedException;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Orient {@link LdapConfigurationSource}.
+ *
+ * @since 3.0
+ */
+@Singleton
+@Named
+public class OrientDBLdapConfigurationSource
+    extends LifecycleSupport
+    implements LdapConfigurationSource
+{
+  private final Provider<DatabaseInstance> databaseInstance;
```

Comment:
Should this be storing the Provider or the underlying DatabaseInstance? It would eliminate the need for the repeated get() pattern to query the Provider if we just did it once in the constructor, but I'm unsure if there are additional lifecycle semantics at play or not...